### PR TITLE
chore: add `crds.podLabels` for helm hook jobs

### PR DESCRIPTION
**Pull request by BryanStenson-okta**
/kind feature
/kind cleanup

This PR adds `crds.podLabels` and uses them (if non-empty) in both hook Job objects.  This is required, for example, to add flexibility when running these CRD jobs in clusters where labels are required for network policy (ie, with Cilium).

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests

### CODE REVIEW

File: manifest_staging/charts/secrets-store-csi-driver/templates/csidriver.yaml [hunk 1]
```diff
@@ -10,5 +10,5 @@ spec:
   - Ephemeral
   {{- if and (semverCompare ">=1.20-0" .Capabilities.KubeVersion.Version) .Values.tokenRequests }}
   tokenRequests:
-    {{- toYaml .Values.tokenRequests | nindent 2}}
+    {{- toYaml .Values.tokenRequests | nindent 2 }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/csidriver.yaml:13]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.windows.enabled}}
+{{- if .Values.windows.enabled }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml:1]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml [hunk 3]
```diff
@@ -132,7 +132,7 @@ spec:
             {{- end }}
             {{- end }}
             {{- if .Values.windows.volumeMounts }}
-              {{- toYaml .Values.windows.volumeMounts | nindent 12}}
+              {{- toYaml .Values.windows.volumeMounts | nindent 12 }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml:135]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml [hunk 4]
```diff
@@ -185,7 +185,7 @@ spec:
         {{- end }}
         {{- end }}
         {{- if .Values.windows.volumes }}
-          {{- toYaml .Values.windows.volumes | nindent 8}}
+          {{- toYaml .Values.windows.volumes | nindent 8 }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver-windows.yaml:188]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.linux.enabled}}
+{{- if .Values.linux.enabled }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml:1]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml [hunk 6]
```diff
@@ -135,7 +135,7 @@ spec:
             {{- end }}
             {{- end }}
             {{- if .Values.linux.volumeMounts }}
-              {{- toYaml .Values.linux.volumeMounts | nindent 12}}
+              {{- toYaml .Values.linux.volumeMounts | nindent 12 }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml:138]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml [hunk 7]
```diff
@@ -185,10 +185,10 @@ spec:
           hostPath:
             path: "{{ $path }}"
             type: DirectoryOrCreate
-        {{- end}}
+        {{- end }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml:188]: nit: remove extra space

File: manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml [hunk 8]
```diff
@@ -185,10 +185,10 @@ spec:
...
         {{- end }}
         {{- if .Values.linux.volumes }}
-          {{- toYaml .Values.linux.volumes | nindent 8}}
+          {{- toYaml .Values.linux.volumes | nindent 8 }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml:191]: nit: remove extra space

**Comment by BryanStenson-okta** [comment]
thanks @nilekhc .  wrt the extra spaces, I was adding them in to follow the convention set by all other uses of `nindent` throughout this chart (ie, https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml#L27 and https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml#L99)

what do you think?

**Comment by nilekhc** [comment]
> thanks @nilekhc . wrt the extra spaces, I was adding them in to follow the convention set by all other uses of `nindent` throughout this chart (ie, https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml#L27 and https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/charts/secrets-store-csi-driver/templates/secrets-store-csi-driver.yaml#L99)
> 
> what do you think?

Ahh. Initially I was thinking to avoid changes not necessary wrt to this. But looks like it's also inconsistent with other places in the chart. Thanks for pointing out. You can ignore my review comments.

/cc @aramase for additional review.

**Comment by aramase** [comment]
/retitle chore: add `crds.podLabels` for helm hook jobs

**Review by aramase — COMMENTED** [review]
Could you add the new parameter in the [configuration](https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/main/manifest_staging/charts/secrets-store-csi-driver#configuration) table?

**Review by aramase — COMMENTED** [review]
Thanks for the PR, @BryanStenson-okta! 

/label tide/merge-method-squash
/approve

**Pull request merged**